lib/path.js

```diff
@@ -698,7 +698,8 @@ const win32 = {
 
 
   dirname: function dirname(path) {
-    assertPath(path);
+    if (typeof path !== 'string')
```

Comment:
It *probably* doesn't matter either way. I just did it this way in case v8 doesn't/can't optimize it by not creating a new string if `path` is already a string.